Formula/astyle.rb

```diff
@@ -1,14 +1,14 @@
 class Astyle < Formula
   desc "Source code beautifier for C, C++, C#, and Java"
   homepage "https://astyle.sourceforge.io/"
-  url "https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.1/astyle_3.1_linux.tar.gz"
-  sha256 "cbcc4cf996294534bb56f025d6f199ebfde81aa4c271ccbd5ee1c1a3192745d7"
+  url "https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.4/astyle-3.4.tar.bz2"
+  sha256 "2bac07a46197da962bf3fb8313373242e2190b7739e4203a703eb89b603aaa07"
   license "MIT"
   head "https://svn.code.sf.net/p/astyle/code/trunk/AStyle"
 
   livecheck do
     url :stable
-    regex(%r{url=.*?/astyle[._-]v?(\d+(?:\.\d+)+)_}i)
+    regex(%r{url=.*?/astyle[._-]v?(\d+(?:\.\d+)+)}i)
```

Comment:
This is picking up some beta versions as well:
```
Matched Versions:
3.4 => #<Version:0x00007fe6e98446d0 @version="3.4", @detected_from_url=false>
3.3.1 => #<Version:0x00007fe6e9844680 @version="3.3.1", @detected_from_url=false>
3.3 => #<Version:0x00007fe6e9844630 @version="3.3", @detected_from_url=false>
3.2.1 => #<Version:0x00007fe6e98445e0 @version="3.2.1", @detected_from_url=false>
3.2 => #<Version:0x00007fe6e9844590 @version="3.2", @detected_from_url=false>
3.1 => #<Version:0x00007fe6e9844540 @version="3.1", @detected_from_url=false>
3.0.1 => #<Version:0x00007fe6e98444f0 @version="3.0.1", @detected_from_url=false>
3.0 => #<Version:0x00007fe6e98444a0 @version="3.0", @detected_from_url=false>
2.06 => #<Version:0x00007fe6e9844450 @version="2.06", @detected_from_url=false>
2.05.1 => #<Version:0x00007fe6e98443d8 @version="2.05.1", @detected_from_url=false>
2.05 => #<Version:0x00007fe6e9844388 @version="2.05", @detected_from_url=false>
2.04 => #<Version:0x00007fe6e9844338 @version="2.04", @detected_from_url=false>
2.03 => #<Version:0x00007fe6e98442e8 @version="2.03", @detected_from_url=false>
```
Version `3.2` is from the `beta` folder, and it isn't present in the `3.2` subfolder. How about this instead?:
```suggestion
    regex(%r{url=.*?/astyle[._-]v?(\d+(?:\.\d+)+)(?:[._-]linux)?\.t}i)
```
With this, we get:
```
Matched Versions:
3.4 => #<Version:0x00007ff6b126abd8 @version="3.4", @detected_from_url=false>
3.3.1 => #<Version:0x00007ff6b126aa48 @version="3.3.1", @detected_from_url=false>
3.3 => #<Version:0x00007ff6b126a9f8 @version="3.3", @detected_from_url=false>
3.2.1 => #<Version:0x00007ff6b126a840 @version="3.2.1", @detected_from_url=false>
3.1 => #<Version:0x00007ff6b126a778 @version="3.1", @detected_from_url=false>
3.0.1 => #<Version:0x00007ff6b126a6b0 @version="3.0.1", @detected_from_url=false>
3.0 => #<Version:0x00007ff6b126a458 @version="3.0", @detected_from_url=false>
2.06 => #<Version:0x00007ff6b126a408 @version="2.06", @detected_from_url=false>
2.05.1 => #<Version:0x00007ff6b126a368 @version="2.05.1", @detected_from_url=false>
2.05 => #<Version:0x00007ff6b126a188 @version="2.05", @detected_from_url=false>
2.04 => #<Version:0x00007ff6b1269ff8 @version="2.04", @detected_from_url=false>
2.03 => #<Version:0x00007ff6b1269f58 @version="2.03", @detected_from_url=false>
```
CC @samford.

Reply:
Makes sense to me 👍

The only thing I would change is using `[._-]` instead of `[_-]` for the optional suffix. It's not the case for `astyle` but I've seen other software use a dot as the delimiter for the suffix. [I went ahead and updated the suggestion above, so it can be easily committed.]

Reply:
Makes sense, thank you both!

Reply:
> The only thing I would change is using `[._-]` instead of `[_-]` for the optional suffix.

I actually went with that first, but changed it to `[_-]` for some reason 😂. Thanks!